src/HospitalModel.H

```diff
@@ -282,6 +250,42 @@ void HospitalModel<AC,ACT,ACTD,A>::treatAgents(AC& a_agents, /*!< Agent containe
             });
             Gpu::synchronize();
 
+            for (int d = 0; d < n_disease; d++) {
+                auto ds_arr = (*a_dstats[d])[mfi].array();
+                ParallelFor( np, [=] AMREX_GPU_DEVICE (int i) noexcept
+                {
+                    if (flag_death_ptr[i] == 1) {
+                        Gpu::Atomic::AddNoRet( &ds_arr(  home_i_ptr[i],
+                                                         home_j_ptr[i],
+                                                         0,
+                                                         DiseaseStats::death),
+                                               1.0_rt );
+                    }
+                    if (flag_hosp_ptr[i] >= 1) {
```

Comment:
Could we re-use the `DiseaseStats` struct here?

Reply:
Done!